docs/modules/ROOT/pages/api/account.adoc

```diff
@@ -0,0 +1,158 @@
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+:snip6: https://github.com/ericnordelo/SNIPs/blob/feat/standard-account/SNIPS/snip-6.md[SNIP-6]
+
+= Account
+
+Reference of interfaces, presets, and utilities related to Account contracts.
+
+TIP: For an overview of Accounts, read our xref:accounts.adoc[Accounts guide].
+
+== Core
+
+[.contract]
+[[ISRC6]]
+=== `++ISRC6++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/account/interface.cairo#L12[{github-icon},role=heading-link]
+
+[.hljs-theme-light.nopadding]
+```javascript
+use openzeppelin::account::interface::ISRC6;
+```
+
+Interface of the ISRC6 standard as defined in the {snip6}.
+
+
+[.contract-index]
+.Functions
+--
+* xref:#ISRC6-\\__execute__[`++__execute__(calls)++`]
+* xref:#ISRC6-\\__validate__[`++__validate__(calls)++`]
+* xref:#ISRC6-is_valid_signature[`++is_valid_signature(hash, signature)++`]
```

Comment:
To me, self shouldn't be part of these API functions, because is not really a part of the interface (similar to the discussion in SRC5). 

It is true that it supposedly shows if the external method writes to storage or not, but this is a development-level convention and is not really enforced in the protocol, meaning that even if self is a snapshot, you can still modify the storage in that method by using the storage_write syscall.

I think that assuming that a function can't write to storage because self is a snapshot is a vulnerability.

Reply:
> I think that assuming that a function can't write to storage because self is a snapshot is a vulnerability.

Mmmm agreed. Perhaps we can revisit when/if this gets fixed because I think, assuming it works as advertised, this is helpful for the user. Maybe this reason alone isn't enough idk

Reply:
After revisiting this a couple of times, I think I'm leaning towards leaving `self` out of interfaces (like ISRC6) API reference, but including it in the modules (like Account), because we are not documenting the external interface, but the API, and self is part of that API when you want to call those module's contents. Also, pure internal methods should be represented differently as you mentioned. I updated the PR to match this already, let me know what do you think

Reply:
+1 to `self` being part of the library's API (even if it's not the contract's ABI)

also, if we ever decide to remove `self` and diverge from the written code of our lib, we should add a clarification for anyone comparing docs vs code